Missions/__AR/11_cont/AI.src

```diff
@@ -2724,20 +2760,33 @@ Every 0$1+0$0.2 do
           if BaseUnderAttack(base,-1) then
                continue;
 
+          // [SHARK] these should count against reservesCount as well! limit them as well.
+
           //The bazookers to attack arrive
-          special_russian_attackers = CreateUnitsWithClass(difficulty + yashin_alive, class_bazooker, side);
+          if russian_reserve_troops > 0 then
+             begin
+                  if logsenabled then
+                     display_strings = ['Spawning russian special troops that attack Gensher! '];
 
-          temp_list = ai_attacking_unit_hex_index[base];
-          for i in special_russian_attackers do
-               begin
-                    PlaceUnitArea(i,russian_reinforcements_area,false);
+                  special_russian_attackers = CreateUnitsWithClassLimited(difficulty + yashin_alive, class_bazooker, side, russian_reserve_troops);
+                  russian_reserve_troops = russian_reserve_troops - sizeof(special_russian_attackers);
 
-                    temp_list = Replace(temp_list,i,1);
-               end;
-          ai_attacking_unit_hex_index = Replace(ai_attacking_unit_hex_index,base,temp_list);
+                  if logsenabled then
+                     display_strings = ['Remaining russian reserves ! ', russian_reserve_troops];
```

Comment:
ditto for all of these `display_strings` calls because this is the only visible line ingame. somewhat useful, but i'd like to see the rest of them really.